I tried to get this to build on OS X (since the repository says MacOS is supported) using `gcc8Stdenv`, but it didn't link successfully:

```nix
{ stdenv, lib, fetchFromGitHub, cmake, pkgconfig, libpthreadstubs, icu }:

stdenv.mkDerivation rec {
  pname = "peaclock";
  version = "0.4.3";

  src = fetchFromGitHub {
    owner = "octobanana";
    repo = pname;
    rev = version;
    sha256 = "1582vgslhpgbvcd7ipgf1d1razrvgpq1f93q069yr2bbk6xn8i16";
  };

  nativeBuildInputs = [ cmake ];
  buildInputs = [ libpthreadstubs icu ];

  cmakeFlags = [ "-DCMAKEBUILD_TYPE=release" ] ++ lib.optional (stdenv.isDarwin)
    "-DCMAKE_CXX_COMPILER=${stdenv.cc}/bin/c++";

  enableParallelBuilding = true;

  meta = with stdenv.lib; {
    description = "A clock, timer, and stopwatch for the terminal.";
    homepage = "https://octobanana.com/software/peaclock";
    license = licenses.mit;
    platforms = [ "x86_64-darwin" "x86_64-linux" "i686-linux" ];
    maintainers = with maintainers; [ djanatyn ];
  };
}
```

```diff
+  peaclock = callPackage ../applications/misc/peaclock {
+    stdenv = if stdenv.isDarwin then gcc8Stdenv else stdenv;
+  };
```

Build output:
```
[ 16%] Building CXX object CMakeFiles/peaclock.dir/src/main.cc.o
[ 50%] Building CXX object CMakeFiles/peaclock.dir/src/ob/string.cc.o
[ 50%] Building CXX object CMakeFiles/peaclock.dir/src/peaclock/tui.cc.o
[ 66%] Building CXX object CMakeFiles/peaclock.dir/src/ob/readline.cc.o
[ 83%] Building CXX object CMakeFiles/peaclock.dir/src/peaclock/peaclock.cc.o
[100%] Linking CXX executable peaclock
ld: warning: option -s is obsolete and being ignored
ld: warning: passed two min versions (10.12.0, 10.12) for platform macOS. Using 10.12.
Undefined symbols for architecture x86_64:
  "std::filesystem::__cxx11::path::has_filename() const", referenced from:
      std::filesystem::__cxx11::path::operator/=(std::filesystem::__cxx11::path const&) in main.cc.o
  "std::filesystem::__cxx11::path::lexically_normal() const", referenced from:
      Tui::mkconfig(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool) in tui.cc.o
  "std::filesystem::__cxx11::path::has_root_directory() const", referenced from:
      std::filesystem::__cxx11::path::operator/=(std::filesystem::__cxx11::path const&) in main.cc.o
  "std::filesystem::__cxx11::path::compare(std::filesystem::__cxx11::path const&) const", referenced from:
      _main in main.cc.o
      Tui::load_config(std::filesystem::__cxx11::path const&) in tui.cc.o
  "std::filesystem::create_directory(std::filesystem::__cxx11::path const&)", referenced from:
      _main in main.cc.o
  "std::filesystem::status(std::filesystem::__cxx11::path const&)", referenced from:
      _main in main.cc.o
      Tui::mkconfig(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool) in tui.cc.o
      Tui::load_config(std::filesystem::__cxx11::path const&) in tui.cc.o
  "std::filesystem::__cxx11::path::_M_split_cmpts()", referenced from:
      std::filesystem::__cxx11::path::path<char [5], std::filesystem::__cxx11::path>(char const (&) [5], std::filesystem::__cxx11::path::format) [clone .isra.274] in main.cc.o
      std::filesystem::__cxx11::path::operator/=(std::filesystem::__cxx11::path const&) in main.cc.o
      std::filesystem::__cxx11::path OB::Parg::get<std::filesystem::__cxx11::path, 0>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in
 main.cc.o
      _main in main.cc.o
      Tui::mkconfig(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool) in tui.cc.o
      Tui::load_config(std::filesystem::__cxx11::path const&) in tui.cc.o
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/peaclock.dir/build.make:163: peaclock] Error 1
make[1]: *** [CMakeFiles/Makefile2:95: CMakeFiles/peaclock.dir/all] Error 2
make: *** [Makefile:149: all] Error 2
builder for '/nix/store/g6wdj74dx7s5wjddz69v7z8782wdnjhy-peaclock-0.4.3.drv' failed with exit code 2
error: build of '/nix/store/g6wdj74dx7s5wjddz69v7z8782wdnjhy-peaclock-0.4.3.drv' failed
```

--- a/maintainers/maintainer-list.nix
+++ b/maintainers/maintainer-list.nix
@@ -1303,10 +1303,12 @@
     name = "Vladimir Serov";
     keys = [
       # compare with https://keybase.io/cab404
-      { longkeyid = "1BB96810926F4E715DEF567E6BA7C26C3FDF7BB3";
+      {
+        longkeyid = "1BB96810926F4E715DEF567E6BA7C26C3FDF7BB3";
         fingerprint = "rsa3072/0xCBDECF658C38079E";
       }
-      { longkeyid = "1EBC648C64D6045463013B3EB7EFFC271D55DB8A";
+      {
+        longkeyid = "1EBC648C64D6045463013B3EB7EFFC271D55DB8A";
         fingerprint = "ed25519/0xB7EFFC271D55DB8A";
       }
     ];
@@ -1431,10 +1433,12 @@
     githubId = 2054509;
     name = "Constantine Evans";
     keys = [
-      { longkeyid = "rsa4096/0xB67DB1D20A93A9F9";
+      {
+        longkeyid = "rsa4096/0xB67DB1D20A93A9F9";
         fingerprint = "32B1 6EE7 DBA5 16DE 526E  4C5A B67D B1D2 0A93 A9F9";
       }
-      { longkeyid = "rsa4096/0x1A1D58B86AE2AABD";
+      {
+        longkeyid = "rsa4096/0x1A1D58B86AE2AABD";
         fingerprint = "669C 1D24 5A87 DB34 6BE4  3216 1A1D 58B8 6AE2 AABD";
       }
     ];
@@ -4373,7 +4377,7 @@
     name = "Jeff Zellner";
   };
   kaction = {
-    name  = "Dmitry Bogatov";
+    name = "Dmitry Bogatov";
     email = "KAction@disroot.org";
     github = "kaction";
     githubId = 44864956;
@@ -7006,7 +7010,7 @@
   purcell = {
     email = "steve@sanityinc.com";
     github = "purcell";
-    githubId =	5636;
+    githubId = 5636;
     name = "Steve Purcell";
   };
   puzzlewolf = {
@@ -7275,10 +7279,12 @@
     github = "rissson";
     githubId = 18313093;
     keys = [
-      { longkeyid = "rsa4096/0xF6FD87B15C263EC9";
+      {
+        longkeyid = "rsa4096/0xF6FD87B15C263EC9";
         fingerprint = "8A0E 6A7C 08AB B9DE 67DE  2A13 F6FD 87B1 5C26 3EC9";
       }
-      { longkeyid = "ed25519/0xBBB7A6801DF1E03F";
+      {
+        longkeyid = "ed25519/0xBBB7A6801DF1E03F";
         fingerprint = "C0A7 A9BB 115B C857 4D75  EA99 BBB7 A680 1DF1 E03F";
       }
     ];
@@ -9749,7 +9755,7 @@
     name = "Asad Saeeduddin";
   };
   ngerstle = {
-    name  = "Nicholas Gerstle";
+    name = "Nicholas Gerstle";
     email = "ngerstle@gmail.com";
     github = "ngerstle";
     githubId = 1023752;

--- a/pkgs/applications/misc/peaclock/default.nix
+++ b/pkgs/applications/misc/peaclock/default.nix
@@ -1,0 +1,24 @@
+{ stdenv, fetchFromGitHub, cmake, libpthreadstubs, icu }:
+
+stdenv.mkDerivation rec {
+  pname = "peaclock";
+  version = "0.4.3";
+
+  src = fetchFromGitHub {
+    owner = "octobanana";
+    repo = pname;
+    rev = version;
+    sha256 = "1582vgslhpgbvcd7ipgf1d1razrvgpq1f93q069yr2bbk6xn8i16";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ libpthreadstubs icu ];
+
+  meta = with stdenv.lib; {
+    description = "A clock, timer, and stopwatch for the terminal";
+    homepage = "https://octobanana.com/software/peaclock";
+    license = licenses.mit;
+    platforms = platforms.unix;
+    maintainers = with maintainers; [ djanatyn ];
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -22755,6 +22755,10 @@ in
     inherit (gst_all_1) gstreamer gst-plugins-base gst-plugins-good gst-libav;
   };
 
+  peaclock = callPackage ../applications/misc/peaclock {
+    stdenv = gccStdenv;
+  };
+
   peek = callPackage ../applications/video/peek { };
 
   pflask = callPackage ../os-specific/linux/pflask {};

